# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/GpuDecoderModule.cpp
# Commit: c5dac46ef11d
# Full Hash: c5dac46ef11d44762d5eb61610e58f4d52aec88d
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - PA: Delete GPUDecoderModule wrapper. r=mattwoodrow
#   
#   Moved support logic into WMFDecoderModule.
#   
#   Depends on D54882
# ==============================================================================

diff -r fca05e774f88 -r c5dac46ef11d dom/media/ipc/GpuDecoderModule.cpp
--- a/dom/media/ipc/GpuDecoderModule.cpp	Tue Oct 20 23:26:33 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,55 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-#include "GpuDecoderModule.h"
-
-#include "RemoteDecoderManagerChild.h"
-#include "RemoteMediaDataDecoder.h"
-#include "RemoteVideoDecoder.h"
-#include "mozilla/StaticPrefs_media.h"
-#include "mozilla/SyncRunnable.h"
-
-namespace mozilla {
-
-using namespace ipc;
-using namespace layers;
-using namespace gfx;
-
-nsresult GpuDecoderModule::Startup() {
-  if (!RemoteDecoderManagerChild::GetManagerThread()) {
-    return NS_ERROR_FAILURE;
-  }
-  return mWrapped->Startup();
-}
-
-bool GpuDecoderModule::SupportsMimeType(
-    const nsACString& aMimeType, DecoderDoctorDiagnostics* aDiagnostics) const {
-  return mWrapped->SupportsMimeType(aMimeType, aDiagnostics);
-}
-
-bool GpuDecoderModule::Supports(const SupportDecoderParams& aParams,
-                                DecoderDoctorDiagnostics* aDiagnostics) const {
-  return mWrapped->Supports(aParams, aDiagnostics);
-}
-
-static inline bool IsRemoteAcceleratedCompositor(KnowsCompositor* aKnows) {
-  TextureFactoryIdentifier ident = aKnows->GetTextureFactoryIdentifier();
-  return ident.mParentBackend != LayersBackend::LAYERS_BASIC &&
-         !ident.mUsingSoftwareWebRender &&
-         ident.mParentProcessType == GeckoProcessType_GPU;
-}
-
-already_AddRefed<MediaDataDecoder> GpuDecoderModule::CreateVideoDecoder(
-    const CreateDecoderParams& aParams) {
-  if (!StaticPrefs::media_gpu_process_decoder() || !aParams.mKnowsCompositor ||
-      !IsRemoteAcceleratedCompositor(aParams.mKnowsCompositor)) {
-    return mWrapped->CreateVideoDecoder(aParams);
-  }
-
-  return RemoteDecoderManagerChild::CreateVideoDecoder(
-      aParams, RemoteDecodeIn::GpuProcess);
-}
-
-}  // namespace mozilla